x-pack/plugins/xpack_main/server/routes/api/v2/telemetry/telemetry.js

```diff
@@ -91,9 +97,21 @@ export function telemetryRoute(server) {
       const config = req.server.config();
       const start = req.payload.timeRange.min;
       const end = req.payload.timeRange.max;
+      const unencrypted = req.payload.unencrypted;
 
       try {
-        return await getTelemetry(req, config, start, end);
+        if (unencrypted) {
+          const isAllowed = unencrypted && isAllowedToViewUnencryptedTelemetryData(req);
```

Comment:
`unencrypted === true` always here.
and the [third check](https://github.com/elastic/kibana/pull/33121/files#diff-89cd25c7be7b127ed0a4d1ccaf48d7cbR110) also makes reader thinking more.
how about to just declare 2 branches?
```js
if (asEncrypted) {
  const usageData = await getTelemetry(req, config, start, end);
  return encryptTelemetry(config, usageData);
} else {
  if(hasAccessToRead(...)){
    const usageData = await getTelemetry(req, config, start, end);
    return usageData 
  }
  return h.response({ code: 'T001' }).code(500);
}
```